src/network/netdev/xfrm.c

```diff
@@ -0,0 +1,81 @@
+/* SPDX-License-Identifier: LGPL-2.1+ */
+
+#include <net/if.h>
+
+#include "netdev/xfrm.h"
+
+static int xfrm_config_verify(
+    NetDev *netdev,
+    const char *filename
+) {
+    Xfrm *x;
+
+    assert(netdev);
+    assert(filename);
+
+    x = XFRM(netdev);
+
+    if (!x->dev) {
+        log_warning("Xfrm NetDev without dev set in %s. Defaulting to 'lo'", filename);
+        if(!((x->dev = strdup("lo")))) {
+            log_error("Could not allocate memory");
+            return -ENOMEM;
+        }
+    }
+
+    if (!x->if_id) {
+        log_warning("Xfrm NetDev without if_id set in %s. Defaulting to '0'", filename);
+        x->if_id = 0;
+    }
+
+    return 0;
+}
+
+static int xfrm_fill_message_create(
+    NetDev *netdev,
+    Link *link,
+    sd_netlink_message *message
+) {
+    int	r;
+    int	ifidx;
+
+    assert(netdev);
+    assert(message);
+
+    Xfrm *x = XFRM(netdev);
+
+    if ((ifidx = if_nametoindex(x->dev)) == 0)
```

Comment:
please dont do two things at once. nothing wrong doing this in two steps: first call if_name_to_index(), then chck the result